app/assets/javascripts/pages/agent-edit-page.js.coffee

```diff
@@ -126,17 +129,49 @@ class @AgentEditPage
       else
         @hideEventCreation()
 
+  buildAce: ->
+    $(".ace-editor").each ->
+      unless $(this).data('initialized')
+        $(this).data('initialized', true)
+        $source = $($(this).data('source')).hide()
+        editor = ace.edit(this)
+        $(this).data('ace-editor', editor)
+        editor.getSession().setTabSize(2)
+        editor.getSession().setUseSoftTabs(true)
+        editor.getSession().setUseWrapMode(false)
+        editor.setTheme("ace/theme/chrome")
+
+        setSyntax = ->
+          syntax = $("[name='agent[options][language]']").val()
```

Comment:
I think using [switch](http://coffeescript.org/#switch) could remove a few lines or at least the local variable.